src/test/objectstore/store_test.cc

```diff
@@ -3414,6 +3414,212 @@ TEST_P(StoreTest, OmapSimple) {
   }
 }
 
+TEST_P(StoreTest, OmapBench) {
+  // Random dictionary generation
+  static const std::string dict =
+      "0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+  mt19937 gen{random_device{}()};
+  uniform_int_distribution<> dist(0, dict.size() - 1);
+
+  // objects
+  int obj_count = 100000;
+  vector<ghobject_t> objects;
+  objects.reserve(obj_count);
+
+  // Keys
+  int keys_per_object = 100;
+  int key_batch_size = 100;
+  int klen = 64;
+  int vlen = 256;
+
+  int r;
+
+  float min = std::numeric_limits<float>::max();
+  float max = std::numeric_limits<float>::min();
+  float avg = std::numeric_limits<float>::min();
+  float total = std::numeric_limits<float>::min();
+
+  // Make the collection
+  coll_t cid;
+  auto ch = store->create_new_collection(cid);
+  {
+    ObjectStore::Transaction t;
+    t.create_collection(cid, 0);
+    cerr << "Creating collection " << cid << std::endl;
+    r = queue_transaction(store, ch, std::move(t));
+    ASSERT_EQ(r, 0);
+  }
+
+  cout << "[OMAPBENCH] *** Setting up objects ***" << std::endl;
+  for (int obj = 0; obj < obj_count; obj++) {
+    string name("object");
+    name += stringify(obj);
+    objects.push_back(ghobject_t(hobject_t(sobject_t(name, CEPH_NOSNAP),
+                                "key", 123, -1, "")));
```

Comment:
IIUC this means the same hash for all the objects. Hence might be the bad performance in unordered_map...